agnpy/utils/conversion.py

```diff
@@ -1,24 +1,37 @@
 # conversion utilities for agnpy
 from astropy.constants import m_e, m_p, h, c, G
 import astropy.units as u
-
+from astropy.constants import e, h, c, m_e, m_p, sigma_T, G
 
 mec2 = m_e.to("erg", equivalencies=u.mass_energy())
 mpc2 = m_p.to("erg", equivalencies=u.mass_energy())
-lambda_c = (h / (m_e * c)).to("cm")  # Compton wavelength
+lambda_c_e = (h / (m_e * c)).to("cm")  # Compton wavelength
+lambda_c_p = (h / (m_p * c)).to("cm")  # Compton wavelength of protons
 # equivalency for decomposing Gauss in Gaussian-cgs units (not available in astropy)
 Gauss_cgs_unit = "cm(-1/2) g(1/2) s-1"
 Gauss_cgs_equivalency = [(u.G, u.Unit(Gauss_cgs_unit), lambda x: x, lambda x: x)]
 # equivalency to transform frequencies to energies in electron rest mass units
-epsilon_equivalency = [
-    (u.Hz, u.Unit(""), lambda x: h.cgs * x / mec2, lambda x: x * mec2 / h.cgs)
-]
-
-
-def nu_to_epsilon_prime(nu, z=0, delta_D=1):
+def epsilon_equivalency(m = m_e):
+    if m == m_e:
+        epsilon_equivalency = [
+            (u.Hz, u.Unit(""), lambda x: h.cgs * x / mec2, lambda x: x * mec2 / h.cgs)
+            ]
+    elif m == m_p:
+        epsilon_equivalency = [
+            (u.Hz, u.Unit(""), lambda x: h.cgs * x / mpc2, lambda x: x * mpc2 / h.cgs)
+            ]
+    else:
+        raise ValueError(
+            f"Provide either the electron or the proton mass. "
+        )
+    return epsilon_equivalency
+
+
+def nu_to_epsilon_prime(nu, z=0, delta_D=1, m = m_e):
```

Comment:
I think this is dangerous. Keeping two different conversions of nu to E depending on the particle can cause problems if somebody would like to plot at the same time electron and proton emission

Reply:
I get your point @jsitarek, but if I am confused by this:
In the synchrotron SED formula there is actually a multiplication by epsilon (A.30 of [our paper](https://www.aanda.org/articles/aa/pdf/2022/04/aa42000-21.pdf)).
@dimaniad6 got a correct SED comparison with Cerruti by multiplying by an epsilon scaled to the proton rest mass energy. How would we obtain the same number if we were to have an epsilon only scaled by the electron mass? We should multiply by the ratio between the two masses, right?

If so, I am even more confused because I remembered there was an inverse cube mass in the synchrotron energy loss. Maybe it is absorbed in the R(x) function and in the epsilon factor?

Reply:
Hi @cosimoNigro 
looking again at my comment, I think I wrote something wrong. Indeed plotting it vs nu is perfectly fine with e- and p together, plotting vs epsilon would be not correct, but as you wrote this is not something that users would do. In the total energy loss for synchrotron radiation of electrons you have sigma_T, which goes ~ 1/ m_e^2 
the differential emission is more complicated, because it includes the bessel function integration with limit being nu/nu_c, where nu_c ~ 1/m_e
see Eq 4.31 and 4.32 in Blumenthal & Gould 1970: 
https://journals.aps.org/rmp/abstract/10.1103/RevModPhys.42.237
(the formula is for emitted spectral power P(nu), which I guess should be EdN / dnu.
there is a formula also for low frequency case (4.34) with P(nu) ~ [1/m_e](nu/2nu_c)^1/3, which would end up as ~m^2/3